src/kademlia/routing_table.cpp

```diff
@@ -758,9 +764,7 @@ routing_table::add_node_status_t routing_table::add_node_impl(node_entry e)
 				// from these nodes, pick the one with the highest RTT
 				// and replace it
 
-				auto k = std::max_element(nodes.begin(), nodes.end()
-					, [](bucket_t::iterator lhs, bucket_t::iterator rhs)
-					{ return lhs->rtt < rhs->rtt; });
+				auto k = std::max_element(nodes.begin(), nodes.end());
```

Comment:
this is not correct. this will compare the iterators, not the values the iterators point to